### PR TITLE
Add new test suite and empty BranchSetup class for branch management

### LINKED ISSUE #17: feat: Set up required branches
# Feature Description

- Verify the existence of required branches: stage and release.
- Implement a minimal set up to allow future customization of branch configurations. Examples include:
  - The option to enable or disable the stage branch.
  - Customizable names for stage and release branches (e.g., using stag instead of stage or production instead of release).
- Ensure branch setup is adaptable:
  - If the user uses only two branches, exclude release candidate tags.
  - If the user uses a single branch, remove development tags.

# Why is this feature needed?

- Proper branch management in essenetial for facilitating releases.

### PR DESCRIPTION
**Pull request by kimseungbin**
Introduce a new `BranchSetup` class and corresponding test spec file. The class currently serves as a placeholder for future branch management functionalities. Initial test cases have been outlined but require implementation.

## Release Notes

- **New Features**
  - Introduced a new test suite for branch setup verification.
  - Added `BranchConfig` interface and `BranchSetup` class for managing branch configurations.
  - Enhanced the `LabelChecker` and `LabelSyncer` classes to utilize a unified repository path structure.
  - Expanded the test suite for `LabelSyncer` with detailed implementations and new test cases.

- **Bug Fixes**
  - Improved error handling in label initialization and synchronization processes.

- **Documentation**
  - Updated issue templates for breaking changes, bug reports, feature requests, and infrastructure tasks to standardize title formats.

- **Chores**
  - Added new dependency `@actions/github` to `package.json`.
  - Modified GitHub Actions workflow to include new input parameters and environment variables.

- Close #17

### CODE REVIEW

File: src/branch-setup.spec.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+		expect(branches).toContain('main')
+		expect(branches).toContain('stage')
+		expect(branches).toContain('release')
+	})
+	describe('verifyBranches', () => {
+		it('should verify existence of all required branches', async () => {
+			const verifyBranches = vi.spyOn(branchSetup, 'verifyBranches')
+			await branchSetup.verifyBranches()
+			expect(verifyBranches).toHaveBeenCalled()
+		})
```
[the remark this answers is not in the document]
Comment (kimseungbin) [reply src/branch-setup.spec.ts:45]: Suggest refactoring

File: src/main.ts [hunk 2]
```diff
@@ -48,7 +48,7 @@ export async function run(): Promise<void> {
 
 	try {
 		// todo Get PR number dynamically
-		const labelSyncer = new LabelSyncer(octokit, owner, repo, 1)
+		const labelSyncer = new LabelSyncer(octokit, repoPath, 1)
```
[the remark this answers is not in the document]
Comment (kimseungbin) [reply src/main.ts:51]: Suggest code

[the remark this answers is not in the document]
Comment (kimseungbin) [reply src/main.ts:51]: @coderabbitai suggest the code

**Comment by kimseungbin** [comment]
@coderabbitai review
